packages/core/src/routes/experience/classes/experience-interaction.ts

```diff
@@ -377,6 +428,11 @@ export default class ExperienceInteraction {
       queries: { users: userQueries, userSsoIdentities: userSsoIdentityQueries },
     } = this.tenant;
 
+    if (EnvSet.values.isDevFeaturesEnabled && !this.captcha.verified && !this.captcha.skipped) {
+      // Check if the captcha is required for the current interaction
+      await this.signInExperienceValidator.guardCaptcha(this.interactionEvent);
+    }
```

Comment:
In the registration flow, the user account will be created at the identification step (see the createUser method above). Need to double-check whether we need to guard the captcha ahead.

You may keep this logic here as the final step check, but put an additional guard at the `/experience/identification` router level.

Reply:
LOG-11058